library/forms.inc

```diff
@@ -141,65 +119,57 @@ function getEncounters($pid, $dateStart = '', $dateEnd = '', $encounterRuleType
         $where .= "form_encounter.pid = ?";
         array_push($arraySqlBind, $pid);
     }
-
         $res = sqlStatement("SELECT distinct encounter FROM $from WHERE $where ORDER by date desc;", $arraySqlBind);
-
     for ($iter=0; $row=sqlFetchArray($res); $iter++) {
         $all[$iter] = $row;
     }
-
         return $all;
 }
-
 function getEncounterDateByEncounter($encounter)
 {
     global $attendant_type;
     $table = $attendant_type == 'pid' ? 'form_encounter' : 'form_groups_encounter';
     // $sql = "select date from forms where encounter='$encounter' order by date";
-    $sql = "select date from $table where encounter='$encounter' order by date";
-    return sqlQuery($sql);
+    $sql = "SELECT date from ? where encounter = ? order by date";
+    return sqlQuery($sql, array($table, $encounter));
 }
-
 function getProviderIdOfEncounter($encounter)
 {
         global $attendant_type;
         $table = $attendant_type == 'pid' ? 'form_encounter' : 'form_groups_encounter';
-        $sql = "select provider_id from $table where encounter=? order by date";
-        $res = sqlQuery($sql, array($encounter));
-        return $res['provider_id'];
+        $sql = "SELECT provider_id FROM ? WHERE encounter = ? ORDER BY date";
+        return sqlQuery($sql, array($encounter, $table))['provider_id'];
```

Comment:
Escape table as discussed above and rec not combining the last 2 lines (harder to follow and will require testing).